sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -1222,20 +1224,31 @@ class SQLContext(@transient val sparkContext: SparkContext)
    * Returns a Catalyst Schema for the given java bean class.
    */
   protected def getSchema(beanClass: Class[_]): Seq[AttributeReference] = {
-    val (dataType, _) = inferDataType(beanClass)
+    val (dataType, _) = SQLContext.inferDataType(TypeToken.of(beanClass))
     dataType.asInstanceOf[StructType].fields.map { f =>
       AttributeReference(f.name, f.dataType, f.nullable)()
     }
   }
 
+}
+
+object SQLContext {
```

Comment:
As long as we are going to break this out into its own object, can we move it to its own file and maybe name it something like `JavaTypeInference`?